app/src/main/java/de/westnordost/streetcomplete/quests/accepts_cash/AddAcceptsCash.kt

```diff
@@ -63,7 +63,7 @@ class AddAcceptsCash(
 
     override fun getTitleArgs(tags: Map<String, String>, featureName: Lazy<String?>): Array<String> {
         val name = tags["name"] ?: tags["brand"]
```

Comment:
Is `tags["brand"]` correct here? A few lines above, it explicitly excluded.

Reply:
Is that because if the hasFeatureName matched a brand (e.g. McDonalds), you don't want it to say:
"Does %1$s (%2$s) accept cash payment?"
"Does McDonalds (McDonalds) accept cash payment?"

But this would be acceptable:
"Does McDonalds (Fast Food Restaurant) accept cash payment?"

Reply:
But that's not what would happen:

Given the following tags:

```
amenity=fast_food
name=McDonald's
brand=McDonald's
```

* `hasFeatureName(tags)` returns `true`
* `tags.containsKey("brand")` returns `true`
* So, `R.string.quest_accepts_cash_title` ("Does %s accept cash payment?") is used
* `getTitleArgs` returns `arrayOf("McDonald's", "Fast Food Restaurant")`
* All in all, this results in "Does *McDonald's* accept cash payment?"

Reply:
Ah yes, sorry I thought hasFeatureName included the NSI presets so would match on the brand as well.

Reply:
Right, there is a mix-up.

In some places 
```kotlin
    private fun hasFeatureName(tags: Map<String, String>?): Boolean =
        tags?.let { featureDictionaryFuture.get().byTags(it).isSuggestion(false).find().isNotEmpty() } ?: false
```
is used to find out if a feature has a "real" preset name.
In other places, a function with the same name is used to find if a feature can be named at all somehow.

I'd be grateful if you cleaned that up. I think in most cases (for the quest title string), what we actually want is the feature name, not the brand/NSI-name. For a few quests, we need "any" name.

Reply:
I already solved this in the frame of a bug I just fixed! I hope you did not start working on it!

Reply:
Oh no, I didn't. But I can do that after merge if that is the last thing that is missing for htis PR